app/javascript/components/UI/Menu.js

```diff
@@ -1,26 +1,105 @@
+    'meme-selection': { src: complete5_10, activeSrc: complete5_10_act, percent: 10 },
+    'selected-giphy-follow': { src: complete15, activeSrc: complete15_act, percent: 15 },
+    'emotion-intensity': { src: complete20, activeSrc: complete20_act, percent: 20 },
+    'productivity-check': { src: complete25, activeSrc: complete25_act, percent: 25 },
+    'productivity-bad-follow-up': { src: complete35, activeSrc: complete35_act, percent: 35 },
+    'causes-to-celebrate': { src: complete45, activeSrc: complete45_act, percent: 45 },
+    'recognition': { src: complete65, activeSrc: complete65_act, percent: 65 },
+    'icebreaker-answer': { src: complete85, activeSrc: complete85_act, percent: 85 },
+    'icebreaker-question': { src: complete90, activeSrc: complete90_act, percent: 90 },
+    'results': { src: complete100, activeSrc: complete100_act, percent: 100 },
+  };
+
+  const getSrcMenu = (lastSegment, activeImg) => {
+    if (segmentsMap[lastSegment]) {
+      const { src, activeSrc, percent } = segmentsMap[lastSegment];
+      return {
+        src: activeImg ? activeSrc : src,
+        percent: percent,
+      };
+    } else {
+      return "";
+    }
+  };
 
-const Menu = ({ percentCompletion, className = ''  }) => {
   return (
     <div className={ `${className}` }>
-      <Dropdown>
+      <Dropdown onClick={handleChangeImg}>
         <Dropdown.Toggle  id='dropdown-stick'>
-        <div  className="menu ">
-          <div  className="m-cover  justify-content-end"></div>
-          <div className='one-line-menu'></div>
-          <div className='one-line-menu'></div>
-          <div className='one-line-menu'></div>
-        </div>
+          <img src={getSrcMenu(lastSegment, activeImg).src} alt='complete' />
         </Dropdown.Toggle>
         <Dropdown.Menu>
-          <Dropdown.Item href="#" ><Button className='btn-item-menu wb1 mx-auto my-auto' onClick={()=>{}}>Manage Profile</Button></Dropdown.Item>
-          <Dropdown.Item href="#" ><Button className='btn-item-menu wb1  mx-auto my-auto' onClick={()=>{}}>Save Draft</Button></Dropdown.Item>
-          <Dropdown.Item href="#" ><Button className='btn-item-menu wb1  mx-auto my-auto' onClick={()=>{}}>Log Out</Button></Dropdown.Item>
+          <Dropdown.Item href="#" >
+            <Button className={`btn-item-menu wb1 mx-auto my-auto${draft || steps.length === 1 ? ' disabled-btn-draft' : ''}`}
+                    disabled={draft || steps.length === 1} onClick={handleSaveDraft}>
+              Save Draft
+            </Button>
+          </Dropdown.Item>
+          <Dropdown.Item href="#" ><Button className='btn-item-menu wb1  mx-auto my-auto' onClick={handleSignOut}>Log Out</Button></Dropdown.Item>
         </Dropdown.Menu>
       </Dropdown>
-      <div className='fs-6 text-primary' style={{margin: 19+"px 0 0 0"}}>{percentCompletion}% complete</div>
+      <div className='fs-6 text-primary' style={{margin: 19+"px 0 0 0"}}>{getSrcMenu(lastSegment).percent }% complete</div>
```

Comment:
you can use style={{margin: "19px 0 0 0"}} here, it is preferable. But even more preferable - is to use class name with predefined styles in scss.